### DISCUSSION TITLE
Cache invalidation without a mutation

### DISCUSSION THREAD

**Discussion by Yuripetusko** [urql-graphql/urql#2985] [Q&A] [answered]
Hello all, I tried to look for a similar questions, but couldn't find exactly what I was looking for, sorry if I missed it.

We have an event driven db, meaning, there are never a mutations that are done from client, as we index db from a blockchain data. Sometimes we need to invalidate cache when we know that a data from indexer was updated, the best way we found so far is to do `requestPolicy: 'network-only'` and calling `refetch`, however this is a rather global option, what would be the best for us is to be able to pass `requestPolicy` to `refetch` function itself, as sometimes we want cached response and other times we want a network-only response. 

Is there a good pattern (we use react hooks) on how to achieve this?

**Comment by kitten** [comment, marked as answer]
If you're manually reexecuting an operation in your UI bindings, all of our bindings allow you to pass a `Partial<OperationContext>`, so you can indeed pass a custom `requestPolicy` per a single reexecute call already 😅✌️

**Reply by Yuripetusko** [reply]
Oh 🤦  We were re-exporting `reExecuteQuery` as `refetch` with custom typings and I didn't notice that. Sorry
